noxfile.py

```diff
@@ -1,26 +1,44 @@
-"""A nox configuration file so that we can build the documentation easily with nox.
-- see the README.md for information about nox.
-- ref: https://nox.thea.codes/en/stable/
-"""
+import glob
+import os.path
+
 import nox
 
 nox.options.reuse_existing_virtualenvs = True
 
-build_command = ["-b", "dirhtml", "docs", "docs/_build/html"]
 
-@nox.session()
+@nox.session(python="3.13", default=False)
 def docs(session):
-    session.install("-r", "docs/requirements.txt")
-    if "live" in session.posargs:
-        # Add relative paths to this if we ever need to ignore them
-        AUTOBUILD_IGNORE = [
-          "docs/_data",
-          "docs/_build",
-        ]
+    """
+    Build the documentation.
+    """
+    sphinx_config_files = glob.glob("./docs/**/conf.py", recursive=True)
+    sphinx_config = sphinx_config_files[0]
+    source_dir = os.path.dirname(sphinx_config)
+    if source_dir.find("source") == -1:
+        output_dir = os.path.join(source_dir, "_build/html")
+    else:
+        output_dir = os.path.join(os.path.dirname(source_dir), "_build/html")
+
+    doc_build_default_args = ["-b", "dirhtml", source_dir, output_dir]
+
+    session.install("-r", os.path.join(source_dir, "requirements.txt"))
+
+    if "live" in session.posargs or (
```

Comment:
I think it makes sense to keep autobuild opt-in, rather than default for most interactive cases. WDYT?

Reply:
The majority of the times that I do a local build, I want to preview the documentation locally.